source/vibe/core/stream.d

```diff
@@ -135,6 +135,12 @@ interface OutputStream {
 interface Stream : InputStream, OutputStream {
 }
 
+/// Interface for streams allowing buffer overrides
+/*interface Buffered {
+	/// Reads at least 1 byte into buf and returns the slice, or 
+	/// returns an empty array if this operation is impossible.
+	ubyte[] readChunk(ubyte[] buf);
+}*/
```

Comment:
This is my fault. This was the initial implementation of a variant of `read` that takes an external buffer, thus enabling TCP connection implementations with no internal buffer. However, there was no clear winner for the best API (w.r.t. `@safe`), so I've removed it for this PR. I can dig up a link to the old discussion. Would definitely be good to get some more ideas.